channeldb/channel.go

```diff
@@ -1287,6 +1296,20 @@ func (c *OpenChannel) UpdateCommitment(newCommitment *ChannelCommitment) error {
 				"revocations: %v", err)
 		}
 
+		// Persist dangling remote updates that need to be restored
+		// after a restart.
+		var b bytes.Buffer
+		err = serializeLogUpdates(&b, danglingRemoteUpdates)
+		if err != nil {
+			return err
+		}
+
+		err = chanBucket.Put(danglingRemoteUpdatesKey, b.Bytes())
```

Comment:
should we delete these at some point (when we sign them), or do we still need them lying around?

Not sure if it is possible, but I'm thinking about a scenario where we receive the remote updates and signature, we `revokeAndAck` and send a new signature. Now we crash and on restart we'll still have these dangling updates, but since we already signed them they will be added to the update log twice.

Reply:
We clear them right below in `AppendRemoteCommitChain`. 

In your example, they will be cleared once we send the new signature.

Reply:
Simplified clear code to just delete the key.

Reply:
Ah, didn't see that a `nil` was serialized. Deleting the key is more clear indeed :)